VellumZero.cs

```diff
@@ -57,27 +57,105 @@ public enum Hook
+                IgnorePatterns = new String[] { },
+                DiscordSync = new DiscordSyncConfig()
+                {
+                    EnableDiscordSync = false,
+                    DiscordToken = "",
+                    DiscordChannel = 0,
+                    DiscordMentions = true,
+                    LatinOnly = false,
+                    DiscordCharLimit = 0
+                },
+                ServerSync = new ServerSyncConfig()
+                {
+                    EnableServerSync = false,
+                    OtherServers = new string[] { },
+                    BusAddress = "127.0.0.1",
+                    BusPort = 8234,
+                    BroadcastChat = true,
+                    DisplayOnlineList = true,
+                    OnlineListScoreboard = "Online",
+                    ServerListScoreboard = "Servers"
+                },
+                VZStrings = new VZTextConfig()
+                {
+                    LogInit = "Initialized",
+                    LogDiscInit = "Starting Discord",
+                    LogDiscLib = "Discord Libraries Loaded",
+                    LogDiscConn = "Discord Connected",
+                    LogDiscDC = "Discord Disconnected",
+                    LogBusConn = "Bus Connected",
+                    LogEnd = "Unloading plugin",
+                    ChatMsg = "(§6{0}§r) <{1}> {2}",
+                    PlayerJoinMsg = "(§6{0}§r) {1} Connected",
+                    PlayerLeaveMsg = "(§6{0}§r) {1} Left",
+                    ServerUpMsg = "(§6{0}§r): §aOnline§r",
+                    ServerDownMsg = "(§6{0}§r): §cOffline§r",
+                    MsgFromDiscord = "(§d{0}§r) [§b{1}§r] {2}",
+                    RestartOneWarn = "The server will restart in {0} {1}",
+                    RestartMinWarn = "§c§lLess than {0} min to scheduled restart!",
+                    RestartSecTitle = "§c{0}",
+                    RestartSecSubtl = "§c§lseconds until restart",
+                    MinutesWord = "minutes",
+                    SecondsWord = "seconds"
+                }
+            };
+        }
+
         public void Initialize(IHost host)
         {
             Host = host;
-            vzConfig = LoadConfiguration();
+
+            // Load the plugin configuration from the hosts run-config
+            vzConfig = host.LoadPluginConfiguration<VZConfig>(this.GetType());
+
+            System.Console.WriteLine(vzConfig.ServerSync.BusAddress);
+
+            // Probably have to rework the plugin system a bit to expose stuff like the world name like in your fork...
+            // In the meantime loading the server.properties once again should do the job :D.
+            // - clarkx86
+
+            using (StreamReader reader = new StreamReader(File.OpenRead(Path.Join(Path.GetDirectoryName(host.RunConfig.BdsBinPath), "server.properties"))))
+                _worldName = Regex.Match(reader.ReadToEnd(), @"^level\-name\=(.+)", RegexOptions.Multiline).Groups[1].Value;
```

Comment:
I'm extending the plugin system upstream so that the host exposes more members and hooks, so this is not necessary in the full release